### ISSUE TITLE
timeout warning in rings section

### ISSUE THREAD

**Issue by pubyun** [haproxy/haproxy#1900]
### Detailed Description of the Problem

haproxy 2.4.19 give warnings though i add timeout parameter.

```
 /usr/sbin/haproxy -c  -f /etc/haproxy/haproxy.cfg
[WARNING]  (1169237) : config : proxy 'logbuffer' has no 'bind' directive. Please declare it as a backend if this was intended.
[WARNING]  (1169237) : config : missing timeouts for proxy 'logbuffer'.
   | While not properly invalid, you will certainly encounter various problems
   | with such a configuration. To fix this, please ensure that all following
   | timeouts are set to a non-zero value: 'client', 'connect', 'server'.
Warnings were found.
Configuration file is valid
```

here is config:

```
ring logbuffer
  description "buffer for logs"
  format rfc5424
  maxlen 1200
  size 32764
  timeout connect         10s
  timeout server          1m
  server logserver 172.18.11.13:5514 check log-proto octet-count

log-forward syslog
  bind xxx.xx.169.49:514
  dgram-bind xxx.xx.169.49:514
  log ring@logbuffer local5
```

### Expected Behavior

no warning

### Steps to Reproduce the Behavior

 /usr/sbin/haproxy -c  -f /etc/haproxy/haproxy.cfg

### What is your configuration?

```haproxy
default
```

### Output of `haproxy -vv`

```plain
HAProxy version 2.4.19-1 2022/09/30 - https://haproxy.org/
```

**Comment by Darlelet** [comment]
Nice catch, thank you for reporting.

I can reproduce using 2.4.19.
2.4.18 is not affected.
2.5, 2.6, 2.7 and master are not affected.

This is a side effect introduced when backporting commit fe0979fbfdbec1e861adbc4d23c9d0b701381f1b to 2.4 branch.

What's interesting is that a commit with the same name was also backported in 2.5.9, but here the behavior remains untouched.

**Comment by Darlelet** [comment]
The reason why 2.5.9 is unaffected by the regression is thanks to 2c04a5a03de50c62fc6520a3cdcd1bab2de794cf (2.5 only).

2c04a5a03de50c62fc6520a3cdcd1bab2de794cf was not backported to 2.4 because it was not mentionned in the commit message.

Maybe we should consider backporting it to 2.4 branch to silence both warnings. (if the patch applies)

Edit: The patch does not apply, PR_CAP_INT flag (required by https://github.com/haproxy/haproxy/commit/2c04a5a03de50c62fc6520a3cdcd1bab2de794cf) is not available in 2.4 (already mentioned in fd66708c43768b7926db1443a602c0939f3a5c01)

**Comment by Darlelet** [comment]
Silencing the warning in > 2.4.19 would require:

- applying 6bb77b9c6465121a5019896982a96a552da54e8e
- amending fd66708c43768b7926db1443a602c0939f3a5c01 (or applying fd3fe4a3335f883d26dc1cb81eb9049ac42b4934 instead) so that PR_CAP_INT is set in sink_setup_proxy()
- applying 2c04a5a03de50c62fc6520a3cdcd1bab2de794cf

**Comment by capflam** [comment]
It should be fixed now in 2.7-DEV. Of course, the bug is only visible on 2.4. So the issue is virtually fixed, waiting for the backports :)

**Comment by pubyun** [comment]
good job, fixed on 2.4.
